test/integration/LoginEndpoints.test.js

```diff
@@ -0,0 +1,127 @@
+import 'babel-polyfill' // Needed because of mocha
+import assert from 'assert'
+
+import StreamrClient from '../../src'
+import config from './config'
+
+const Web3 = require('web3')
+
+const web3 = new Web3()
+
+describe('LoginEndpoints', () => {
+    let client
+
+    const createClient = (opts = {}) => new StreamrClient({
+        url: config.websocketUrl,
+        restUrl: config.restUrl,
+        apiKey: 'tester1-api-key',
+        autoConnect: false,
+        autoDisconnect: false,
+        ...opts,
+    })
+
+    beforeAll(() => {
+        client = createClient()
+    })
+
+    describe('Challenge generation', () => {
+        it('should retrieve a challenge', () => client.getChallenge({
+            address: 'some-address',
+        })
+            .then((challenge) => {
+                assert(challenge)
+                assert(challenge.id)
+                assert(challenge.challenge)
+                assert(challenge.expires)
+            }))
+    })
+
+    async function assertThrowsAsync(fn, regExp) {
```

Comment:
Hmm. Surely there's a built-in and elegant way to test that a promise is rejected? For such a common case you shouldn't need to roll your own?

Reply:
Yes there is `assert.rejects()` in Node 10, but we use Node 8... I didn't really create this function, I found it online as a workaround for versions prior to Node 10. I don't know if there exists a more elegant way to do it.